CHANGELOG.md

```diff
@@ -1,14 +1,406 @@
 # Changelog
 
-Note: The changelog is severely out of date, and will be updated *later*.
-
-These changes have been made since the current 2.2.0 release. If you build the project as-is, it will include all of them.
 
 > What's the point of going through all the trouble of the debug process if you're going to go and fiddle with the game afterwards…?
 >
 > — Satoru Iwata, "[Iwata Asks – Pokémon HeartGold Version & SoulSilver Version](https://www.nintendo.co.uk/Iwata-Asks/Iwata-Asks-Pokemon-HeartGold-Version-SoulSilver-Version/Iwata-Asks-Pokemon-HeartGold-Version-SoulSilver-Version/1-Just-Making-The-Last-Train/1-Just-Making-The-Last-Train-225842.html)"
 
 
+Polished Crystal v3.1.1 brings with it a lot of changes. What started as a Dex expansion, through three years of development it has evolved into a strikingly different game from the v3.0.0-beta release with numerous added features from newer generations, a plethora of code optimizations, and many quality of life improvements. Ultimately, the game aims to feel like a modern Pokemon game with retro graphics. The rule of thumb is to assume that things work like modern games, and if it doesn't, it is either a bug or an oversight.
+
+What follows is a short description alongside a list of the changes made to several iterations of Polished Crystal from the last stable release (v3.0.0-beta). This is not an exhaustive list, but covers all of the major changes, and a majority of the minor ones. There are content spoilers.
+
+Note: There are some changes listed from v2.2.0 to v3.0.0-beta as well. These are not in any way exhaustive, and likely don't list several major changes.
+
+## v3.1.1
+
+This is a hotfix release that removes one bug that caused the overworld graphics to glitch out badly when hatching an egg.
+
+## v3.1.0
+
+This is the newest stable version release which finished a number of branches that were under development and added them to v3.0.0, along with a few other small additions.
+
+### Additions
+- Bloodmoon Ursaluna, with its signature ability "Mind's Eye", is now available. It can be obtained by leveling up Ursaring in Murky Swamp. 332 Pokemon now complete the pokedex.
+- Added the much requested NPC to point the players in the direction of the Go-Goggles.
+- The Candy Jar can now be obtained in the Goldenrod Underground. Exp Candies can be bought as well.
+- Support for the GB Printer was added so that players can print the Deploma.
+- The Galarian Bird event is completed. They are now accessible through the Pokemon Go trainers in the old Rocket Hideout in Celadon City.
+- Cherrygrove Tree in the bay now has text without the items used for the G-Bird event.
```

Comment:
```suggestion
```
This was fixed before the release, and wasn't a bug carried over from v3.0.0, so we don't need to talk about it; it just falls under the Galarian Bird event.